### PR TITLE
ISSUE-137 Added option to OAS to format report like Terrain

### PR DESCRIPTION
**Pull request by marklim**
I've modified OAS to add an option to export the report in the same format as Terrain, i.e. I-statements per row. Scouts per column. This will make it easier to transfer the statements back into Terrain after having printed them out for say a camp

### CODE REVIEW

File: Services/ReportService.cs [hunk 1]
```diff
@@ -1017,7 +1017,139 @@ public IWorkbook GenerateAttendanceReportWorkbook(AttendanceReportModel attendan
...
+                rowNumber++;
+                if (pdrText != plan.InputTitle)
+                {
+                    pdrText = plan.InputTitle;
+                    sheet.Range[rowNumber, columnNumber].Text = plan.InputTitle;
+                    sheet.Range[rowNumber, columnNumber].CellStyle.Color = GetInputTitleColour(plan.InputTitle);
+                    sheet.Range[rowNumber, columnNumber].BorderAround();
+                    sheet.Range[rowNumber, columnNumber].CellStyle.Rotation = 90;
+                    sheet.Range[rowNumber, columnNumber].CellStyle.Font.Bold = true;
+                    sheet.Range[rowNumber, columnNumber].CellStyle.HorizontalAlignment = ExcelHAlign.HAlignCenter;
```
Comment (NomisNostab) [inline Services/ReportService.cs:1037]: Maybe Plan, Do and Review are at 0 and centred across the rows.
```
                    sheet.Range[rowNumber, columnNumber].CellStyle.Rotation = 0;
                    sheet.Range[rowNumber, columnNumber].CellStyle.Font.Bold = true;
                    sheet.Range[rowNumber, columnNumber].CellStyle.VerticalAlignment = ExcelVAlign.VAlignCenter;

```

File: Services/ReportService.cs [hunk 2]
```diff
@@ -1017,7 +1017,139 @@ public IWorkbook GenerateAttendanceReportWorkbook(AttendanceReportModel attendan
...
+            columnNumber++;
+            sheet.Range[titleRow + 1, columnNumber + 1].FreezePanes();
+            rowNumber = titleRow;
+            foreach (var groupedAnswer in groupedAnswers.OrderBy(ga => ga.Key))
+            {
+                columnNumber++;
+                sheet.Range[rowNumber, columnNumber].Text = groupedAnswer.Key;
+                sheet.Range[rowNumber, columnNumber].BorderAround();
+                sheet.Range[rowNumber, columnNumber].CellStyle.Font.Bold = true;
+
```
Comment (NomisNostab) [inline Services/ReportService.cs:1069]: Can we rotate the names 90 so the page doesn't get too wide with about 20 kids.
```
                sheet.Range[rowNumber, columnNumber].CellStyle.Rotation = 90;
                sheet.Range[rowNumber, columnNumber].CellStyle.HorizontalAlignment = ExcelHAlign.HAlignCenter;

```

**Review by NomisNostab — COMMENTED** [review]
Just some thoughts on name and PDR rotation.

File: Services/ReportService.cs (hunk 2 above)
Comment (marklim) [reply Services/ReportService.cs:1071]: Done

**Pull request merged**